src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -698,10 +830,13 @@ private void ParseTargetElement(Target target, ILoggingConfigurationElement targ
 
                 return true;
             }
+
             return false;
         }
 
-        private bool ParseCompoundTarget(Dictionary<string, ILoggingConfigurationElement> typeNameToDefaultTargetParameters, string name, ILoggingConfigurationElement childElement,
+        private bool ParseCompoundTarget(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)